watchman.c

```diff
@@ -52,6 +58,11 @@ watchman_sock_connect(const char *sockname, struct timeval timeout, struct watch
 {
     struct sockaddr_un addr = { };
 
+    use_bser_encoding = getenv("GIT_USE_JSON_PROTOCOL") == NULL;
```

Comment:
Let's keep git-specific stuff out of here -- change these var names to LIBWATCHMAN_ (even though that's a bit annoying).